src/Sulu/Bundle/PageBundle/Resources/js/views/Form/toolbarActions/TemplateToolbarAction.js

```diff
@@ -26,6 +27,22 @@ export default class TemplateToolbarAction extends AbstractFormToolbarAction {
             this.resourceFormStore.setType(this.defaultTemplate);
         }
 
+        if (this.router.attributes.parentId && this.webspace) {
+            ResourceRequester.get('pages', {
```

Comment:
I didn't think of that, that is a little bit unfortunate :see_no_evil: But I also don't another way than loading the parent page, as this information is not really available anywhere else :thinking:

However, especially with that line of code I would include a check if an ID is set on the surrounding `if` as well... And we should make sure that this request ist not made multiple times, I am not completely sure how often this function is called during runtime...

Reply:
I've set the parentPage as soon as the request is completed, the next time this function will be called there's a check if parentPage is already set or not.

Reply:
Did you check if that was really necessary first? I.e. if the request was sent multiple times in some circumstances?

Reply:
It was sent multiple times, though only one time with a non-empty webspace...

Reply:
Okay, we've put some thoughts into this, and this is probably not the way to go. Loading this data in here, means that we would have to send another request in another `ToolbarAction`, if we also need data from the parent there.

In order to avoid this request, we could do that directly in the [`handleItemAdd` function of our `PageList`](https://github.com/sulu/sulu/blob/f39900f/src/Sulu/Bundle/PageBundle/Resources/js/views/PageList/PageList.js#L152). We already have there everything we need: The `Webspace` object and the `id` of the parent page. Then we could add this to the object being passed to the `navigate` call as `defaultTemplate`, and just use `this.router.attributes.defaultTemplate` in the `TemplateToolbarAction`.

WDYT?